src/spyglass/common/common_interval.py

```diff
@@ -53,20 +57,144 @@ def insert_from_nwbfile(cls, nwbf, *, nwb_file_name):
+        **kwargs,
+    ) -> Tuple[bool, dict]:
+        """Check if the interval list already exists.
+
+        Parameters
+        ----------
+        key : dict
+            The key for the IntervalList table.
+        approx_name : str, optional
+            A string that should be in the interval list name. If None, just
+            search by nwb_file_name. Defaults to None. If `True`, will attempt
+            to generalize the name by replacing numbers with '%'.
+        time_limit : float, optional
+            The maximum time to search for an existing interval list before
+            inserting a new one. Defaults to 5.0.
+
+        Returns
+        -------
+        tuple
+            The first element is a boolean indicating if the interval list
+            already exists. The second element is the key of the existing
+            interval list if it exists, or the original key if it does not.
+        """
+
+        start_time = time()
+        query = self & {"nwb_file_name": key.get("nwb_file_name")}
+        if approx_name is True:
+            approx_name = self._generalize_name(key["interval_list_name"])
+        if approx_name:
+            query &= f"interval_list_name LIKE '%{approx_name}%'"
+
+        if not query:
+            return False, key
+
+        candidates = query.fetch(as_dict=True)
+        random_shuffle(candidates)
+        for candidate in candidates:
+            if time() - start_time > time_limit:
+                return False, key
+            if not np.equal(candidate["valid_times"], key["valid_times"]).all():
+                continue
+            logger.info(
+                "Interval valid times exist: "
+                + f"{key['interval_list_name']} -> "
+                + f"{candidate['interval_list_name']}"
+            )
+            return True, candidate
+
+    def cautious_insert1(
+        self,
+        key: dict,
+        approx_name: str = None,
+        time_limit: float = 5.0,
+        *args,
+        **kwargs,
+    ) -> dict:
+        """Check if the interval list already exists before inserting it.
+
+        If the interval list already exists, return the existing interval list.
+        Will insert if search time exceeds `time_limit`.
+
+        Parameters
+        ----------
+        key : dict
+            The primary key for the IntervalList table.
+        approx_name : str, optional
+            A string that should be in the interval list name. If None, just
+            search by nwb_file_name. Defaults to None. If `True`, will attempt
+            to generalize the name by replacing numbers with '%'.
+        time_limit : float, optional
+            The maximum time to search for an existing interval list before
+            inserting a new one. Defaults to 5.0.
+        """
+        exists, new_key = self._check_existing(
+            key, approx_name, time_limit, *args, **kwargs
+        )
+
+        if not exists:
+            self.insert1(key, *args, **kwargs)
```

Comment:
should this do a check if the exact key is already present whether the interval lists match?  Would resolve some of the comments I have in LFP and LFPBand below